cdap-cli/src/main/java/co/cask/cdap/cli/command/AbstractSetPreferencesCommand.java

```diff
@@ -91,6 +96,55 @@ protected String determinePattern(String action) {
     throw new RuntimeException("Unrecognized element type: " + type.getShortName());
   }
 
+  protected String determineNewPattern(String action) {
+
+    switch (action) {
+      case "set":
+        return determineNewSetPatternHelper();
+      case "load":
+        return determineNewLoadPatternHelper();
+    }
+    throw new RuntimeException("Unrecognized element type: " + type.getShortName());
+  }
+
+  private String determineNewSetPatternHelper() {
```

Comment:
similar comment, think it's better to have a normal name for this and rename the existing one to 'old' or 'deprecated'